source/Coop.Core/Server/Services/Sync/Handlers/PeerQueueOverloadedHandler.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Common;
+using Common.Messaging;
+using Common.Network;
+using Coop.Core.Server.Services.Time.Handlers;
+using Coop.Core.Server.Services.Time.Messages;
+using GameInterface.Services.GameDebug.Messages;
+using GameInterface.Services.Heroes.Enum;
+using LiteNetLib;
+using System;
+using System.Collections.Generic;
+namespace Coop.Core.Server.Services.Sync.Handlers
+{
+    internal class PeerQueueOverloadedHandler : IHandler
+    {
+        private static readonly long POLL_INTERVAL = 100;
+
+        private readonly IMessageBroker messageBroker;
+        private readonly INetwork network;
+
+        private readonly Poller poller;
+        private readonly HashSet<NetPeer> overloadedPeers;
+
+        private readonly TimeHandler timeHandler;
+
+        private TimeControlEnum originalSpeed;
+
+        public PeerQueueOverloadedHandler(
```

Comment:
Could you create a policy similar to [PlayersLoadingPolicy](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/cb111e423f3fda881f81876b1212ff98b2162c4c/source/Coop.Core/Server/Services/Time/Handlers/TimeHandler.cs#L82) to prevent unpausing during overload. Remember to [add](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/cb111e423f3fda881f81876b1212ff98b2162c4c/source/Coop.Core/Server/Services/Time/Handlers/TimeHandler.cs#L37C10-L37C10) and [remove](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/cb111e423f3fda881f81876b1212ff98b2162c4c/source/Coop.Core/Server/Services/Time/Handlers/TimeHandler.cs#L46) (but in dispose instead) the policy here as well.